Module 5/Ex0ToEx5 V2/Sources/Intern.cpp

```diff
@@ -0,0 +1,70 @@
+#include "Intern.hpp"
+#include "PresidentialPardonForm.hpp"
+#include "RobotomyRequestForm.hpp"
+#include "ShrubberyCreationForm.hpp"
+#include "memory"
+
+std::unique_ptr<Form> Intern::makeForm(const std::string &formName, const std::string &target)
+{
+  std::map<std::string, FormCreator> formCreators = {
+      {"shrubbery creation", &Intern::createShrubberyCreationForm},
+      {"robotomy request", &Intern::createRobotomyRequestForm},
+      {"presidential pardon", &Intern::createPresidentialPardonForm},
+      {"mutant pig termination", &Intern::createMutantPigTerminationForm}};
+
+  auto it = formCreators.find(formName);
+  if (it != formCreators.end()) {
+    std::unique_ptr<Form> form = (this->*(it->second))(target);
+    std::cout << "Intern creates a " << form->getName() << " Form (s.grade "
+              << form->getGradeSign() << ", ex.grade "
+              << form->getGradeExecute() << ") targeted on " << target
+              << " (Unsigned)" << std::endl;
+    return form;
+  } else {
+    throw UnknownFormException();
+  }
+}
+
+std::unique_ptr<Form> Intern::createShrubberyCreationForm(const std::string &target) const
+{
+  return std::make_unique<ShrubberyCreationForm>(target);
+}
+
+std::unique_ptr<Form> Intern::createRobotomyRequestForm(const std::string &target) const
+{
+  return std::make_unique<RobotomyRequestForm>(target);
+}
+
+std::unique_ptr<Form> Intern::createPresidentialPardonForm(const std::string &target) const
+{
+  return std::make_unique<PresidentialPardonForm>(target);
+}
+
+std::unique_ptr<Form> Intern::createMutantPigTerminationForm(const std::string &target) const
+{
+  class MutantPigTerminationForm : public Form {

```

Comment:
Why this class, and why here ?
I didn't find any mention of the MutantPigTerminationForm in the exercise, only an exemple, and it did not prevent you from creating a new file.
It would be better to have it as the other forms, in a separate file.